### DISCUSSION TITLE
symbol names search exact match

### DISCUSSION THREAD

**Discussion by dcarlt** [microsoft/vscode-cpptools#8064] [Q&A]
I'm using an extension that is using vscode.executeWorkspaceSymbolProvider api calls to look for exact matches to symbols in my c++ project as part of its functionality.

I have some member functions that I would like to link into using this api driven symbol search but the symbol search is not finding them, I think because the match has to be exact.

I notice when I do a direct symbol search (not using the above api call, rather using ctrl+t) of the form"#ClassName::FunctionName" I get no results returned. It does find a match if I do a search of the form  "#FunctionName ClassName" (some fuzzy match?) but the latter search does not return anything when done using the api call.

Is there a way to find what the exact name of the symbol is so I can make sure I use the correct search term for the, it appears, more strict search performed by the api call? 

I'm also open to modifying the extension I'm using if there is a better api call to use for getting symbols but seems like that shouldn't be necessary (how is vscode.executeWorkspaceSymbolProvider call different from doing # symbol searches?)

Thanks!

**Comment by sean-mcmanus** [comment]
You're hitting a VS Code issue. Our extension correctly returns results for `#className::FunctionName` but VS Code is filtering them out -- it's being tracked by https://github.com/microsoft/vscode/issues/98125 .
